tests/test_web_protocol.py

```diff
@@ -239,7 +239,7 @@ def test_srv_keep_alive_disable(srv):
 
 async def test_bad_method(srv, loop, buf):
     srv.data_received(
-        b'!@#$ / HTTP/1.0\r\n'
+        b':BAD; / HTTP/1.0\r\n'
```

Comment:
Is there a test case for having multiple spaces in Request-Line? (>2)
What about test with non-ascii?

Reply:
If we want to have such tests, we would need [hypothesis](https://github.com/HypothesisWorks/hypothesis/tree/master/hypothesis-python) to generate such malformed requests. That's a bit bigger story than fixing a single regexp, but certainly will improve tests quality a lot.

Reply:
If somebody wants to write hypothesis tests -- you are welcome (requires another PR)

Reply:
Sure, just wanted to point out what I personally faced with when redesigning another HTTP parser :)